internal/controllers/clusterextension_controller_test.go

```diff
@@ -931,51 +497,15 @@ func verifyConditionsInvariants(t *testing.T, ext *ocv1alpha1.ClusterExtension)
 	}
 }
 
-func TestGeneratedBundleDeployment(t *testing.T) {
-	test := []struct {
-		name              string
-		clusterExtension  ocv1alpha1.ClusterExtension
-		bundlePath        string
-		bundleProvisioner string
-	}{
-		{
-			name: "when all the specs are provided.",
-			clusterExtension: ocv1alpha1.ClusterExtension{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "test-bd",
-					UID:  types.UID("test"),
-				},
-				Spec: ocv1alpha1.ClusterExtensionSpec{
-					InstallNamespace: "test-ns",
-				},
-			},
-			bundlePath:        "testpath",
-			bundleProvisioner: "foo",
-		},
-	}
-
-	for _, tt := range test {
-		fakeReconciler := &controllers.ClusterExtensionReconciler{}
-		objUnstructured := fakeReconciler.GenerateExpectedBundleDeployment(tt.clusterExtension, tt.bundlePath, tt.bundleProvisioner)
-		resultBundleDeployment := &rukpakv1alpha2.BundleDeployment{}
-		require.NoError(t, runtime.DefaultUnstructuredConverter.FromUnstructured(objUnstructured.Object, resultBundleDeployment))
-		// Verify the fields that have are being taken from cluster extension.
-		require.Equal(t, tt.clusterExtension.GetName(), resultBundleDeployment.GetName())
-		require.Equal(t, tt.bundlePath, resultBundleDeployment.Spec.Source.Image.Ref)
-		require.Equal(t, tt.bundleProvisioner, resultBundleDeployment.Spec.ProvisionerClassName)
-		require.Equal(t, tt.clusterExtension.Spec.InstallNamespace, resultBundleDeployment.Spec.InstallNamespace)
-	}
-}
-
 func TestClusterExtensionUpgrade(t *testing.T) {
 	cl, reconciler := newClientAndReconciler(t)
 	ctx := context.Background()
 
 	t.Run("semver upgrade constraints enforcement of upgrades within major version", func(t *testing.T) {
+		t.Skip("Skip and replace with e2e for BundleDeployment")
```

Comment:
Ideally our controller should be easily unit testable. I am fine with it being a follow up for now, but I don't think we should be replacing unit tests with e2e tests

Reply:
See #874.

Reply:
See #879 for making `reconcile()` more testable, but this test will no longer be skipped after #874 is merged.